quantus/metrics/robustness/max_sensitivity.py

```diff
@@ -258,75 +260,105 @@ def __call__(
             softmax=softmax,
             device=device,
             model_predict_kwargs=model_predict_kwargs,
+            batch_size=batch_size,
             **kwargs,
         )
 
-    def evaluate_instance(
+    def evaluate_batch(
         self,
         model: ModelInterface,
-        x: np.ndarray,
-        y: np.ndarray,
-        a: np.ndarray,
-        s: np.ndarray,
-    ) -> float:
+        x_batch: np.ndarray,
+        y_batch: np.ndarray = None,
+        a_batch: np.ndarray = None,
+        s_batch: np.ndarray = None,
+        perturb_func: Callable = None,
+        perturb_func_kwargs: Dict = None,
+        nr_samples: int = None,
+        similarity_func: Callable = None,
+        norm_numerator: Callable = None,
+        norm_denominator: Callable = None,
```

Comment:
not sure about making all of these arguments to the evaluate_batch / evaluate_instance method...
since they are parameters that are supposed to define the metric, and be set (only) in __init__.

but this is not that important for now. I think we should discuss which hyperparameters go where again anyway sometime soon, especially wrt normalisation/abs/aggregation.

Reply:
yes that's also a point i wanted to discuss.
i personally prefer to have these additional metric variables as method arguments instead of attributes.
this way i can already gauge from the method signature what will be actually used during computation.
but that's just my preferred way of doing this, and attributes will work as well.

as this is more or less hidden to the user who basically uses only `__init__()` and `__call__()`, the discussion to handle this consistently could be deferred to a point after the merge (and create an issue to not forget).

Reply:
we should also discuss which of the methods of a `Metric` should be marked as private.

Maybe `general_preprocess()`, `custom_preprocess()`, `custom_postprocess()`, `evaluate_instance()` and so on should be renamed to `__general_preprocess()`, `__custom_preprocess()`, `__custom_postprocess()`, `__evaluate_instance()` respectively.

This way the API html doc-page of each specific metric implementation won't be crowded by methods the actual users don't care about.
The base metric implementation could still include the documentation for these private methods and will be then the goto-reference for developers of new metrics.

Reply:
Also, adding the variables as arguments to the evaluate_batch method just duplicates a lot of documentation, cause everywhere we pass it, it has to be re-documented.

Reply:
Daniel can we, for this PR, use the self.similarity_func instead of passing similarity_func? Unless there is a big benefit of the approach that I cannot see

Reply:
Agree on this: _"Maybe general_preprocess(), custom_preprocess(), custom_postprocess(), evaluate_instance() and so on should be renamed to __general_preprocess(), __custom_preprocess(), __custom_postprocess(), __evaluate_instance() respectively."_

but scope for a separate PR

Reply:
I see your points and have now removed all the stuff regarding additional arguments in the evaluate methods.